### ISSUE TITLE
1.4 Delete then re-create same CBLView causes view not found error

### ISSUE THREAD

**Issue by brendand** [couchbase/couchbase-lite-ios#1872]
I have these CBLViews that I create and then delete and then re-create with the same name. The views represent a set of records in a form. So I may delete the form at some point and delete the corresponding records view.

However, I've found that I'm getting an assertion failure whenever I try to access the view that was deleted then re-created.

```
WARNING: Exception caught in CBLDatabase transaction:
	Assertion failed: View 'records-frm-6b388a2138d64d8fa422bfe8403400e1' not found in database
	5   CouchbaseLite                       0x0000000101062fee __39-[CBL_SQLiteViewStorage updateIndexes:]_block_invoke + 8540
	6   CouchbaseLite                       0x0000000101082ba1 -[CBL_SQLiteStorage inTransaction:] + 370
	7   CouchbaseLite                       0x0000000101060d66 -[CBL_SQLiteViewStorage updateIndexes:] + 224
	8   CouchbaseLite                       0x000000010107f855 -[CBLView updateIndexes:] + 78
	9   CouchbaseLite                       0x000000010107f462 -[CBLView updateIndex] + 57
	10  CouchbaseLite                       0x000000010107fac3 -[CBLView totalRows] + 27
	11  Tap Forms 5                         0x0000000100183686 -[TFCategory forms] + 2550
	12  Tap Forms 5                         0x0000000100183ab2 -[TFCategory formsWithEntries] + 114
	13  Tap Forms 5                         0x000000010027f65a -[TFDatabaseDocument loadCategoriesWithEntries] + 1242
	14  Tap Forms 5                         0x0000000100274465 -[TFDatabaseDocument preloadModelObjects] + 293
	15  Tap Forms 5                         0x00000001002793ab __31-[TFDatabaseDocument refreshUI]_block_invoke + 1435
	16  libdispatch.dylib                   0x00000001013f0cfe _dispatch_call_block_and_release + 12
	17  libdispatch.dylib                   0x00000001013e778c _dispatch_client_callout + 8
	18  libdispatch.dylib                   0x00000001013f5ac4 _dispatch_main_queue_callback_4CF + 362
	19  CoreFoundation                      0x00007fff818d4bc9 __CFRUNLOOP_IS_SERVICING_THE_MAIN_DISPATCH_QUEUE__ + 9
	... {at MYReportException:43}
14:30:55.782| WARNING: CouchbaseLite: Failed to rebuild views (records-frm-6b388a2138d64d8fa422bfe8403400e1): 594 {at -[CBL_SQLiteViewStorage updateIndexes:]:532}
14:30:55.782| WARNING: Error 594 updating index of CBLView[db-5764caa159814fc7aec294c25c0ec60d/records-frm-6b388a2138d64d8fa422bfe8403400e1] {at -[CBLView updateIndex]:264}
```

Here's my code that's causing the exception:

```
[form recordsQuery];
NSString *viewName = [NSString stringWithFormat:@"records-%@", form.document.documentID];
CBLView *recordsView = [self.database existingViewNamed:viewName];
if (recordsView) {
	form.recordCount = @(recordsView.totalRows);
	recordCountUpdated = YES;
}
```

The assertion failure is right on the `form.recordCount = @(recordsView.totalRows);` line.

So the database thinks the view exists because it gets past the `if (recordsView)` condition, but then fails saying it doesn't exist.

Is there another way that I should be deleting the views? I just call `[recordsView deleteView]` to delete the view and it gets re-created within the above call to `[forms recordsQuery]`.

--------
* Version:   1.4.1
* Client OS: macOS 10.12.6
* Server:    n/a

**Comment by pasin** [comment]
> Is there another way that I should be deleting the views? 

I don't know any other way.

What does `[form recordsQuery]` do? Can you generalize CBL calls so that I can create a test for it?

**Comment by brendand** [comment]
`[form recordsQuery]` just creates the view, sets up the MAPBLOCK and returns a CBLQuery.

I'll see if I can put together a test app to demonstrate the problem.

**Comment by djpongh** [comment]
@brendand, any chance to put together that sample app? That will really help us debug.

**Comment by brendand** [comment]
I don't have the time right at the moment to build a sample app. I have to remember when I was able to trigger this bug in my app first :) I think it happened when I deleted a view, then restored a file which contained information that would cause the view to be re-created with the same name. Ya, that must have been it. I haven't tested this in quite a while, so I'll have to see if it's still an issue first. Thanks for checking back in.
